proposals/0000-allow-constraint-type-synonyms-in-deriving-heads.md

```diff
@@ -0,0 +1,89 @@
+---
+author: Grigorii Gerasev
+date-accepted: ""
+ticket-url: ""
+implemented: ""
+---
+
+This proposal is [discussed at this pull request](https://github.com/ghc-proposals/ghc-proposals/pull/597).
+
+# Proposal title
+
+Allow constraints type synonyms in deriving heads again.
+
+## Motivation
+
+Most of Haskell projects do a lot of `stock` and `anyclass` deriving
+for repeating classes. Most of time there is some default list
+of classes to be defived.
+
+Making type synonym for that seems like the obvious thing to do,
+jet it does not work.
+
+It worked at some moment for any instances,
+but was disabled due to wrong behaviour in some cases:
+https://gitlab.haskell.org/ghc/ghc/-/issues/13267
+
+Seems like only phantom param case is relevant for deriving instances case.
+
+## Proposed Change Specification
+
+Accept type synonyms in deriving heads, by rewriting it to regular constraints.
+
+## Examples
+
+```
+type Synonym x = (Show x, Eq x)
+
+-- Should works exactly the same
+derive instance ... => (Synonym x)
+derive instance ... => (Show x, Eq x)
```

Comment:
correction:
```suggestion
## Examples

```haskell
type Synonym x = (Show x, Eq x)

-- Should works exactly the same
deriving instance ... => (Synonym x)
deriving instance ... => (Show x, Eq x)
```

Reply:
Thanks